### PR TITLE
Add basic functionality to timelines

### LINKED ISSUE #8: Implement infinite scrolling and pull to refresh
- [x] I know that I'm requesting a feature for Codename Starlight (SwiftUI) and not for Hyperspace Desktop (JS).

**Is your feature request related to a problem? Please describe.**

Right now we only display a maximum of 20 statuses, which is good for testing, but you obviously can't do anything else. 

**Describe the solution you'd like**

We need to add both pull to refresh and infinite scrolling so that if a user pulls down the list, the latest posts are fetched and shown, and if the user scrolls down to the bottom, older posts are fetched and shown.

**Additional context**

There's no additional context.

### PR DESCRIPTION
**Pull request by alexmodrono**
**Changes Overview**  
This PR makes the following changes:

- Now you can do infinite scrolling on timelines to see older posts.
- Now you can pull to refresh to get the latest posts.

While this pr was mostly focused on improving timelines, it also:
- Improved attachments:
    - Introduced Galeries: attachments are now shown in galleries that allow you to see them in a fancier way.
        - > NOTE: Galleries show up to 3 attachments max.
- Improved the status view.
- Simplified ProfileView.
- Introduced more bugs we'll need to fix later.

**Does this PR fix, close, or implement any issues?**  
- [x] This PR closes, fixes, or implements the following issues.

- Implements #8 
- Somehow contributes to #6 

**Known issues**
- ~~If you tap below the author's profile view, the app will crash. As far as I know, it is caused because the Avatar's nav link is triggered at the same time as the other thread nav link, so UIKit crashes because there are "multiple active destinations".~~ **Fixed by 29d1749b43fe389e39a91e9273c2189edba28050**
- You still can't view attachments in full screen.
- If you go to a profile that isn't yours, the banner and the avatar won't load.

**Pending for review**  
@hyperspacedev/swift-ui

### CODE REVIEW

**Review by alicerunsonfedora — CHANGES_REQUESTED** [review]
The profile page appears to be fine, but I am running into this fatal error when tapping on a post from the Network tab:

```
2020-08-23 21:32:39.929392-0400 Hyperspace[830:147862] WF: _WebFilterIsActive returning: NO
2020-08-23 21:32:40.117865-0400 Hyperspace[830:147862] [assertion] Error acquiring assertion: <Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}>
2020-08-23 21:32:40.117994-0400 Hyperspace[830:147862] [ProcessSuspension] 0x11decfa80 - ProcessAssertion: Failed to acquire RBS Background assertion 'WebProcess Background Assertion' for process with PID 837, error: Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}
2020-08-23 21:32:40.119401-0400 Hyperspace[830:147862] [assertion] Error acquiring assertion: <Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}>
2020-08-23 21:32:40.119549-0400 Hyperspace[830:147862] [ProcessSuspension] 0x11decfac0 - ProcessAssertion: Failed to acquire RBS Suspended assertion 'WebProcess Suspended Assertion' for process with PID 837, error: Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}
Fatal error: UIKitNavigationBridge: multiple active destinations: file SwiftUI, line 0
2020-08-23 21:32:40.292135-0400 Hyperspace[830:147862] Fatal error: UIKitNavigationBridge: multiple active destinations: file SwiftUI, line 0
```

I am running the latest iOS/macOS betas with the latest Xcode.

**Comment by alexmodrono** [comment]
> The profile page appears to be fine, but I am running into this fatal error when tapping on a post from the Network tab:
> 
> ```
> 2020-08-23 21:32:39.929392-0400 Hyperspace[830:147862] WF: _WebFilterIsActive returning: NO
> 2020-08-23 21:32:40.117865-0400 Hyperspace[830:147862] [assertion] Error acquiring assertion: <Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}>
> 2020-08-23 21:32:40.117994-0400 Hyperspace[830:147862] [ProcessSuspension] 0x11decfa80 - ProcessAssertion: Failed to acquire RBS Background assertion 'WebProcess Background Assertion' for process with PID 837, error: Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}
> 2020-08-23 21:32:40.119401-0400 Hyperspace[830:147862] [assertion] Error acquiring assertion: <Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}>
> 2020-08-23 21:32:40.119549-0400 Hyperspace[830:147862] [ProcessSuspension] 0x11decfac0 - ProcessAssertion: Failed to acquire RBS Suspended assertion 'WebProcess Suspended Assertion' for process with PID 837, error: Error Domain=RBSAssertionErrorDomain Code=2 "Specified target process does not exist" UserInfo={NSLocalizedFailureReason=Specified target process does not exist}
> Fatal error: UIKitNavigationBridge: multiple active destinations: file SwiftUI, line 0
> 2020-08-23 21:32:40.292135-0400 Hyperspace[830:147862] Fatal error: UIKitNavigationBridge: multiple active destinations: file SwiftUI, line 0
> ```
> 
> I am running the latest iOS/macOS betas with the latest Xcode.

Gotcha! Fixed by 29d1749b43fe389e39a91e9273c2189edba28050 :D

Instead of having multiple NavigationLinks, what I did was to reuse one, but with different destinations depending on the activation place, so now the two NavigationLinks don't get triggered at the same time.

**Review by alicerunsonfedora — APPROVED** [review]
I did notice an issue with the threads when activated from the profile page (causes a blank page, see screenshots), but this issue isn't app-breaking and can be fixed as we go along. This issue appears to resolve itself when switching to another app and then coming back to the thread.

On initial load:
Screenshot 2020-08-26 at 15 37 16

After switching apps:
Screenshot 2020-08-26 at 15 38 28

File: .swiftlint.yml [hunk 1]
```diff
@@ -5,6 +5,9 @@ excluded:
 opt_in_rules:
   - empty_count
 
+type_body_lenght:
```
Comment (alicerunsonfedora) [inline .swiftlint.yml]: This also appears to be misspelled.

Comment (alexmodrono) [reply .swiftlint.yml]: Fixed by 7338fe00e356705a3c96de51af5dce8af181fdc5

**Comment by alexmodrono** [comment]
Alright so I'm merging this into `data/client` then.

**Pull request merged**